### PR TITLE
calling to endpoints from feed

### PR DESCRIPTION
**Pull request by uiucjcwhite**

### CODE REVIEW

File: src/main/java/com/socialfeed/domain/DataProvider.java [hunk 1]
```diff
@@ -71,26 +72,64 @@
...
+				break;
 			
-			while (resultSet.next())
-			{
-				ids.add(resultSet.getString("id"));
+			case DataProvider.GROUP_MEMBERSHIP_TABLE:
+				String getGroupIds = urlName + EndpointConstants.CORE_PORT + "/" + EndpointConstants.GROUP_MEMBERSHIP +
+					String.format("?user_id={0}&type=Member", id);
+				url = new URL(getGroupIds);
+				conn = (HttpURLConnection) url.openConnection();
```
Comment (aliu899) [inline src/main/java/com/socialfeed/domain/DataProvider.java]: can we refactor some of these common things?, like url = new.. conn , data, disconnect

**Review by aliu899 — APPROVED** [review]

**Review by aliu899 — APPROVED** [review]

File: src/main/java/com/socialfeed/domain/DataProvider.java [hunk 2]
```diff
@@ -71,35 +87,123 @@
...
+				for (UserConnection userConnection: userConnections)
+				{
+					System.out.println("User Id " + userConnection.getId());
+					System.out.println(String.format("Adding User2 %s", userConnection.getUser2()));
+					ids.add(userConnection.getUser2());
+				}
+				
+//				for (UserConnection userConnection: jacksonUserConnections)
+//				{
+//					System.out.println(String.format("Jackson Adding User2 %s", userConnection.getUser2()));
```
Comment (aliu899) [inline src/main/java/com/socialfeed/domain/DataProvider.java:121]: remove these comments?

File: src/main/java/com/socialfeed/domain/DataProvider.java [hunk 3]
```diff
@@ -71,35 +87,123 @@
...
-				resultSet = sqlConnection.getTableRowByColumnName(tableName, "user2", id);
-				DataProvider.updateIdSet(resultSet, ids);
+				url = new URL(getUserIds + String.format("?user2=%s&type=Friend", id));
+				conn = (HttpURLConnection) url.openConnection();
+				data = DataProvider.getJsonDataFromConnection(conn);
+				//userConnections = g.fromJson(data, new TypeToken<List<UserConnection>>(){}.getType());
+				userConnections = objectMapper.readValue(data,  typeFactory.constructCollectionType(ArrayList.class, UserConnection.class));
+				for (UserConnection userConnection: userConnections)
+				{
+					System.out.println("User 2 Id " + userConnection.getId());
```
Comment (aliu899) [inline src/main/java/com/socialfeed/domain/DataProvider.java:132]: will have to convert these to logs if they are important

Comment (uiucjcwhite) [reply src/main/java/com/socialfeed/domain/DataProvider.java:132]: Yeah we do.

File: src/main/java/com/socialfeed/domain/DataProvider.java [hunk 4]
```diff
@@ -109,58 +213,213 @@
...
+		case DataProvider.GROUP:
+			returnEntity = new Group(
+					id,
+					entity.get(Group.NAME),
+					entity.get(Group.DESCRIPTION),
+					InvitePolicyEnum.valueOf(entity.get(Group.INVITE_POLICY)),
+					entity.get(Group.CITY),
+					entity.get(Group.STATE),
+					GeoUtility.Location.parseLocation(entity.get("exact_location")),
+					ViewPrivacy.valueOf(entity.get(Group.VIEW_PRIVACY)),
```
Comment (aliu899) [inline src/main/java/com/socialfeed/domain/DataProvider.java:331]: going to have to update with new fields

File: src/main/java/com/socialfeed/domain/DataProvider.java [hunk 5]
```diff
@@ -109,58 +213,213 @@
...
+		
+		return entities;
+	}
+	
+	@Async
+	public static Future<ArrayList<Event>> getClosestEvents(Location location) throws IOException, ParseException
+	{
+		String getClosestEvents = EndpointConstants.HOST + ":" + EndpointConstants.CORE_PORT + EndpointConstants.EVENT;
+		URL url = new URL(getClosestEvents + String.format("?view_privacy=Public&exact_location=%s&limit=50&sort=distance&orderby=ASC", location.toString()));
+		System.out.println(url.toString());
```
Comment (aliu899) [inline src/main/java/com/socialfeed/domain/DataProvider.java:384]: necessary?

Comment (aliu899) [reply src/main/java/com/socialfeed/domain/DataProvider.java:384]: for logs that is

File: src/main/java/com/socialfeed/domain/DataProvider.java [hunk 6]
```diff
@@ -109,58 +213,213 @@
...
-				exactLocation = GeoUtility.Location.parseLocation(resultSet.getString("exact_location"));
-				entities.add(new Group(id, exactLocation));
-				break;
+				paramList += "&";
 			}
 		}
 		
-		return entities;
+		URL url = new URL(getEventsFromGroups + paramList);
+		System.out.println(url.toString());
```
Comment (aliu899) [inline src/main/java/com/socialfeed/domain/DataProvider.java:416]: ""

File: src/main/java/com/socialfeed/domain/FeedData.java [hunk 7]
```diff
@@ -48,26 +52,38 @@ public void addInterest(String interest)
...
 		if (position >= this.feedItems.size())
 		{
 			position = this.feedItems.size();
 		}
 		
-		if (index != -1)
+		if (index == -1)
 		{
-			this.feedItems.add(index, entity);
+			System.out.println(String.format("Inserting Entity %s at postion %d", entity.getId(), position));
```
Comment (aliu899) [inline src/main/java/com/socialfeed/domain/FeedData.java:70]: ""

File: src/main/java/com/socialfeed/domain/FeedData.java [hunk 8]
```diff
@@ -48,26 +52,38 @@ public void addInterest(String interest)
...
+			this.feedItems.add(position, entity);
 		} 
+		// Only PROMOTE the ranking of an entity
+		else if (position < index)
+		{
+			this.feedItems.remove(index);
+			this.feedItems.add(position, entity);
+			System.out.println(String.format("Removing Entity %s from index %d and moving to postion %d",
```
Comment (aliu899) [inline src/main/java/com/socialfeed/domain/FeedData.java:78]: ""

**Review by aliu899 — CHANGES_REQUESTED** [review]
1. can you pull updates and update Group fields.
2. GroupRelationship entity has info for childgroups and parent groups. can you use this for suggested groups in the feed?
3. are all the system.outs for debugging or metrics? can you distinguish between the two in the print statements

**Pull request merged**